frontend/lib/identification/activation_workflow/activation_code_parser.dart

```diff
@@ -69,25 +65,10 @@ class IdentificationQrContentParser {
       throw QRCodeMissingExpiryException();
     }
 
-    final cardType = CardType.values[bavarianCardType.value]; // FIXME: Insecure mapping
     if (!activationCode.hasTotpSecret()) {
       throw QrCodeFieldMissingException("totpSecret");
     }
-    String? base32TotpSecret;
-    try {
-      base32TotpSecret = base32.encode(Uint8List.fromList(activationCode.totpSecret));
-    } on Exception catch (_) {
-      throw QRCodeInvalidTotpSecretException();
-    }
 
-    final cardDetails = CardDetails(
-      cardInfo.fullName,
-      activationCode.pepper,
-      expirationDay,
-      cardType,
-      cardInfo.extensions.extensionRegion.regionId,
-      base32TotpSecret,
-    );
-    _cardDetailsModel.setCardDetails(cardDetails);
+    _activationCodeModel.setCode(qrcode.dynamicActivationCode);
```

Comment:
I'm not yet sure why this function sets global scope where the very similar `DynamicVerifyCode parseQRCodeContent(String rawBase64Content)` does not do so.